Brokerages/ISymbolMapper.cs

```diff
@@ -0,0 +1,65 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+namespace QuantConnect.Brokerages
+{
+    /// <summary>
+    /// Provides the mapping between Lean symbols and brokerage specific symbols.
+    /// </summary>
+    public interface ISymbolMapper
+    {
+        /// <summary>
+        /// Converts a Lean symbol instance to a brokerage symbol
+        /// </summary>
+        /// <param name="symbol">A Lean symbol instance</param>
+        /// <returns>The brokerage symbol</returns>
+        string GetBrokerageSymbol(Symbol symbol);
+
+        /// <summary>
+        /// Converts a brokerage symbol to a Lean symbol instance
+        /// </summary>
+        /// <param name="brokerageSymbol">The brokerage symbol</param>
+        /// <returns>A new Lean Symbol instance</returns>
+        Symbol GetLeanSymbol(string brokerageSymbol);
+
+        /// <summary>
+        /// Returns the security type for a brokerage symbol
+        /// </summary>
+        /// <param name="brokerageSymbol">The brokerage symbol</param>
+        /// <returns>The security type</returns>
+        SecurityType GetBrokerageSecurityType(string brokerageSymbol);
+
+        /// <summary>
+        /// Returns the security type for a Lean symbol
+        /// </summary>
+        /// <param name="leanSymbol">The Lean symbol</param>
+        /// <returns>The security type</returns>
+        SecurityType GetLeanSecurityType(string leanSymbol);
+
+        /// <summary>
+        /// Checks if the symbol is supported by the brokerage
+        /// </summary>
+        /// <param name="brokerageSymbol">The brokerage symbol</param>
+        /// <returns>True if the brokerage supports the symbol</returns>
+        bool IsKnownBrokerageSymbol(string brokerageSymbol);
```

Comment:
Does this assume that the symbol mapper implementations have a complete listing of all symbols? This seems unrealistic, especially for `InteractiveBrokersBrokerage` which will need to support all security types and even different markets.  Further I think it assumes that symbols are unique within a brokerage, but I fear this is not the case. Consider the equity symbol `AAPL` which can trade in various markets (USA, Mexico, ect..).

It appears that these functions are only used in downloaders and tests. Perhaps they can be concrete implementations on `FxcmSymbolMapper` and `OandaSymbolMapper` but not on `ISymbolMapper`.